src/network/network.cpp

```diff
@@ -700,6 +707,40 @@ bool Network::BeginClient(const char* host, unsigned short port)
 	});
 
 	mode = NETWORK_MODE_CLIENT;
+	char path[MAX_PATH];
+	platform_get_user_directory(path, NULL);
+	char keyPath[MAX_PATH] = "";
+	safe_strcat(keyPath, path, MAX_PATH);
+	safe_strcat(keyPath, gConfigNetwork.player_name, MAX_PATH);
+	safe_strcat(keyPath, ".privkey", MAX_PATH);
+	if (!platform_file_exists(keyPath)) {
+		log_warning("generating key... this may take a while");
```

Comment:
If there's not enough entropy available in the system, openssl may need to wait until more gets available.